conans/client/cmd/uploader.py

```diff
@@ -72,22 +73,29 @@ def _run_upload(self, pattern, force=False, all_packages=False, confirm=False,
                 upload = self._user_io.request_boolean(msg)
             if upload:
                 self._upload(conan_ref, force, all_packages, retry, retry_wait, skip_upload,
-                             integrity_check)
+                             integrity_check, no_overwrite)
 
     def _upload(self, conan_ref, force, all_packages, retry, retry_wait, skip_upload,
-                integrity_check):
+                integrity_check, no_overwrite):
         """Uploads the recipes and binaries identified by conan_ref"""
         if not force:
             self._check_recipe_date(conan_ref)
+
+        if no_overwrite == "all":
+            self._check_recipe_different(conan_ref)
+        elif no_overwrite == "recipe":
+            self._check_recipe_already_uploaded(conan_ref)
```

Comment:
You don't have to check that it is already uploaded, the check is that it is different to the existing one in the server.
So the recipe check is identical for "all" and "recipe". It is just the binaries check that is different: "all" raises, "recipe" allows binary overwrite.